internal/parser/grammar.md

```diff
@@ -88,9 +88,16 @@ The parser directly implements the following grammar.
     FunctionBodyExpression         = Block | Expression .
     Block                          = "{" StatementList "}" .
     ExpressionList                 = [ Expression { "," Expression } ] .
+    ObjectBody                     = identifier ObjectBodySuffix .
+                                   | string_lit PropertyListSuffix .
+    ObjectBodySuffix               = "with" PropertyList
+                                   | PropertyListSuffix .
+    PropertyListSuffix             = PropertySuffix { "," PropertyList } .
     PropertyList                   = [ Property { "," Property } ] .
-    Property                       = identifier [ ":" Expression ]
-                                   | string_lit ":" Expression .
+    Property                       = identifier PropertySuffix
+                                   | string_lit PropertySuffix .
+    PropertySuffix                 = [ ":" Expression ] 
```

Comment:
I think the alternate operator is unnecessary here since `[ a ]` means "this expression is optional" so I think `[ a ]` already means what you put here.